src/features/VisualNovel/pages/MediaPage/MediaPage.tsx

```diff
@@ -0,0 +1,37 @@
+import React, { VFC } from 'react';
+import { useParams } from 'react-router';
+import { Image } from '@chakra-ui/react';
+import { useVisualNovelQuery } from '../../queries';
+import { useSettingsContext } from '../../../../providers';
+
+import cls from './MediaPage.module.css';
+import { VisualNovelScreenshot } from '../../../../models/visualNovel';
+
+/** Media page tab, contains vn screenshots. */
+export const MediaPage: VFC = () => {
+  const { id } = useParams();
+  const { isLoading, error, data } = useVisualNovelQuery(id);
+
+  const settingsContext = useSettingsContext();
+
+  /** Filter screenshots by nsfw flag. */
+  const filterPredicate = (screen: VisualNovelScreenshot): boolean => (settingsContext.showNsfwImages ? true : !screen.isNsfw);
+
+  if (isLoading) {
+    return <>Loading...</>;
+  }
+
+  if (error) {
+    return <>{`An error has occurred: ${error.message}`}</>;
+  }
+
```

Comment:
I think in the future we will need to replace this with our custom components for loading and errors. Maybe add TODO comment for this section?

Reply:
We are working on it right now.

Reply:
We are working on it right now